shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/ddlgenerator/dialect/postgres/PostgresColumnPropertiesLoader.java

```diff
@@ -0,0 +1,317 @@
+        if (typname.indexOf(schema + "\".") > 0) {
+            return typname.substring(schema.length() + 3);
+        }
+        if (typname.indexOf(schema + ".") > 0) {
+            return typname.substring(schema.length() + 1);
+        }
+        return typname;
+    }
+    
+    private String getFullTypeValue(final String name, final String schema, final String length, final String array) {
+        if ("char".equals(name) && "pg_catalog".equals(schema)) {
+            return "\"char\"" + array;
+        } else if ("time with time zone".equals(name)) {
+            return "time" + length + " with time zone" + array;
+        } else if ("time without time zone".equals(name)) {
+            return "time" + length + " without time zone" + array;
+        } else if ("timestamp with time zone".equals(name)) {
+            return "timestamp" + length + " with time zone" + array;
+        } else if ("timestamp without time zone".equals(name)) {
+            return "timestamp" + length + " without time zone" + array;
+        } else {
+            return name + length + array;
+        }
+    }
+    
+    private String checkTypmod(final Integer typmod, final String name) {
+        String length = "(";
+        if ("numeric".equals(name)) {
+            int len = (typmod - 4) >> 16;
+            int prec = (typmod - 4) & 0xffff;
+            length += String.valueOf(len);
+            length += "," + prec;
+        } else if ("time".equals(name) || "timetz".equals(name) || "time without time zone".equals(name) || "time with time zone".equals(name)
+                || "timestamp".equals(name) || "timestamptz".equals(name) || "timestamp without time zone".equals(name) || "timestamp with time zone".equals(name)
+                || "bit".equals(name) || "bit varying".equals(name) || "varbit".equals(name)) {
+            int len = typmod;
+            length += String.valueOf(len);
+        } else if ("interval".equals(name)) {
+            int len = typmod & 0xffff;
+            length += len > 6 ? "" : String.valueOf(len);
+        } else if ("date".equals(name)) {
+            length = "";
+        } else {
+            int len = typmod - 4;
+            length += String.valueOf(len);
+        }
+        if (!length.isEmpty()) {
+            length += ")";
+        }
+        return length;
+    }
+    
+    private String parseTypeName(final String name) {
+        String result = name;
+        boolean isArray = false;
+        if (result.endsWith("[]")) {
+            isArray = true;
+            result = result.substring(0, result.lastIndexOf("[]"));
+        }
+        int idx = result.indexOf("(");
+        if (idx > 0 && result.endsWith(")")) {
+            result = result.substring(0, idx);
+        } else if (idx > 0 && result.startsWith("time")) {
+            int endIdx = result.indexOf(")");
+            if (1 != endIdx) {
+                Pattern pattern = Pattern.compile("(\\(\\d+\\))");
```

Comment:
@tuichenchuxin Can we extract Pattern.compile to global field?

Reply:
Thanks. Done.